crates/driver/src/infra/api/routes/settle/mod.rs

```diff
@@ -8,41 +8,93 @@ use {
             observe,
         },
     },
+    tokio::sync::{mpsc, oneshot},
     tracing::Instrument,
 };
 
 pub(in crate::infra::api) fn settle(router: axum::Router<State>) -> axum::Router<State> {
     router.route("/settle", axum::routing::post(route))
 }
 
+pub(in crate::infra::api) struct QueuedSettleRequest {
+    state: State,
+    req: dto::Solution,
+    response_sender: oneshot::Sender<Result<(), competition::Error>>,
+}
+
+pub(in crate::infra::api) fn create_settle_queue_sender(
+    queue_size: usize,
+) -> mpsc::Sender<QueuedSettleRequest> {
+    let (sender, mut receiver) = mpsc::channel::<QueuedSettleRequest>(queue_size);
+
+    // Spawn the background task to process the queue
+    tokio::spawn(async move {
+        while let Some(queued_request) = receiver.recv().await {
+            let QueuedSettleRequest {
+                state,
+                req,
+                response_sender,
+            } = queued_request;
+
+            let auction_id = req.auction_id;
+            let solver = state.solver().name().to_string();
+            if state.eth().current_block().borrow().number >= req.submission_deadline_latest_block {
+                if let Err(err) =
+                    response_sender.send(Err(competition::Error::QueueAwaitingDeadlineExceeded))
+                {
+                    tracing::error!(
+                        ?err,
+                        "settle deadline exceeded. unable to return a response"
+                    );
+                }
+                continue;
+            }
+
+            let result = async move {
+                observe::settling();
+                let result = state
+                    .competition()
+                    .settle(
+                        req.auction_id,
+                        req.solution_id,
+                        req.submission_deadline_latest_block,
+                    )
+                    .await;
+                observe::settled(state.solver().name(), &result);
+                result.map(|_| ()).map_err(Into::into)
+            }
+            .instrument(tracing::info_span!("/settle", solver, auction_id))
+            .await;
+
+            if let Err(err) = response_sender.send(result) {
+                tracing::error!(?err, "Failed to send /settle response");
+            }
+        }
+    });
+
+    sender
+}
+
 async fn route(
     state: axum::extract::State<State>,
     req: axum::Json<dto::Solution>,
 ) -> Result<(), (hyper::StatusCode, axum::Json<Error>)> {
-    let auction_id = req.auction_id;
-    let solver = state.solver().name().to_string();
-
-    let handle_request = async move {
-        observe::settling();
-        let result = state
-            .competition()
-            .settle(
```

Comment:
I would imagine the API code to be mostly untouched by this change. It would still just call `competition.settle()` but the submission logic now internally manages a queue.